### PR TITLE
remove dtype from to_array

### PR DESCRIPTION
**Pull request by MarcoGorelli**
similar to https://github.com/data-apis/dataframe-api/pull/293

### CODE REVIEW

File: spec/API_specification/dataframe_api/dataframe_object.py [hunk 1]
```diff
@@ -903,26 +903,26 @@ def drop_nulls(
...
+        - Int64() -> 'int64'
+        - UInt8() -> 'uint8'
+        - UInt16() -> 'uint16'
+        - UInt32() -> 'uint32'
+        - UInt64() -> 'uint64'
+        - Float32() -> 'float32'
+        - Float64() -> 'float64'
+
+        and multiple columns' dtypes are combined according to the Array API's
+        type promotion rules.
```
Comment (kkraus14) [inline spec/API_specification/dataframe_api/dataframe_object.py:925]: Maybe link off to the type promotion rules? https://data-apis.org/array-api/latest/API_specification/type_promotion.html

Also, they don't define rules for mixed integer and floating-point types and specify it as implementation defined. Are we good with that as well?

Comment (MarcoGorelli) [reply spec/API_specification/dataframe_api/dataframe_object.py:925]: good call, have added

yeah that sounds fine to me, you OK with it too?

File: spec/API_specification/dataframe_api/dataframe_object.py [hunk 2]
```diff
@@ -813,26 +813,26 @@ def drop_nulls(
...
```
Comment (kkraus14) [inline spec/API_specification/dataframe_api/dataframe_object.py:835]: Did we want to link to the array API docs on these type promotion rules?

**Review by kkraus14 — APPROVED** [review]

File: spec/API_specification/dataframe_api/dataframe_object.py (hunk 2 above)
Comment (MarcoGorelli) [reply spec/API_specification/dataframe_api/dataframe_object.py]: sorry, forgot to push, have done so now

**Pull request merged**